src/web/App.js

```diff
@@ -112,41 +112,45 @@ App.prototype.bake = function(step) {
-                    app.dishStr = Utils.stripHtmlTags(response.result, true);
-                } else {
-                    app.dishStr = response.result;
-                }
-
-                app.progress = response.progress;
-                app.manager.recipe.updateBreakpointIndicator(response.progress);
-                app.manager.output.set(response.result, response.type, response.duration);
-
-                // If baking took too long, disable auto-bake
-                if (response.duration > app.options.autoBakeThreshold && app.autoBake_) {
-                    app.manager.controls.setAutoBake(false);
-                    app.alert("Baking took longer than " + app.options.autoBakeThreshold +
-                        "ms, Auto Bake has been disabled.", "warning", 5000);
-                }
-            })
-            .catch(function(err) {
-                console.error("Chef's promise was rejected, this should never occur");
-            });
+        // This timeout is so the UI has time to update the baking status
+        // before any blocking operations delay it until after the operation.
+        setTimeout(function() {
+            app.chef.bake(
+                app.getInput(),        // The user's input
+                app.getRecipeConfig(), // The configuration of the recipe
+                app.options,           // Options set by the user
+                app.progress,          // The current position in the recipe
+                step                   // Whether or not to take one step or execute the whole recipe
+            )
+                .then(function(response) {
+                    app.setBakingStatus(false);
+
+                    if (!response) return;
+                    if (response.error) app.handleError(response.error);
+
+                    app.options = response.options;
+
+                    if (response.type === "html") {
+                        app.dishStr = Utils.stripHtmlTags(response.result, true);
+                    } else {
+                        app.dishStr = response.result;
+                    }
+
+                    app.progress = response.progress;
+                    app.manager.recipe.updateBreakpointIndicator(response.progress);
+                    app.manager.output.set(response.result, response.type, response.duration);
+
+                    // If baking took too long, disable auto-bake
+                    if (response.duration > app.options.autoBakeThreshold && app.autoBake_) {
+                        app.manager.controls.setAutoBake(false);
+                        app.alert("Baking took longer than " + app.options.autoBakeThreshold +
+                            "ms, Auto Bake has been disabled.", "warning", 5000);
+                    }
+                })
+                .catch(function(err) {
+                    console.error("Chef's promise was rejected, this should never occur");
+                });
+        }, 10);
```

Comment:
I believe @n1474335  doesn't plan on keeping this in.

There is some discussion above about this problem. A possible solution is web-workers which @n1474335 wants to work on at a later date, I think your input could be quite valuable here (there is almost certainly something that we have missed).

The catch statement referenced here should never be called. The throwing away of the stack trace occurs in [Recipe.js](https://github.com/tlwr/CyberChef/blob/388d02ea6840395a5439c8e9451e3285b6014cf9/src/core/Recipe.js#L151). `Recipe.js` definitely needs some TLC (on my todo) (both on my bad promises code front, and operationally (manging of stack traces into CyberChef UI focused error messages)).

It is probably prudent, now that the UI is being decoupled from the Chef/Recipe/Operation logic, that we change how we handle errors and stack traces (to work better with node).

I.e. we preserve a stack trace (and operation metadata) and handle the error message formatting (from the actual trace) in `App.js`